app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -488,20 +484,19 @@ void OpenXRInputSource::UpdateHaptics(ControllerDelegate &delegate)
     CHECK_XRCMD(applyHapticFeedback(mHapticAction, duration, XR_FREQUENCY_UNSPECIFIED, pulseIntensity));
 }
 
-bool OpenXRInputSource::GetHandTrackingInfo(const XrFrameState& frameState, XrSpace localSpace, const vrb::Matrix& head) {
+bool OpenXRInputSource::GetHandTrackingInfo(XrTime predictedDisplayTime, XrSpace localSpace, const vrb::Matrix& head) {
     if (OpenXRExtensions::sXrLocateHandJointsEXT == XR_NULL_HANDLE || mHandTracker == XR_NULL_HANDLE)
         return false;
 
     // Update hand locations
     XrHandJointsLocateInfoEXT locateInfo { XR_TYPE_HAND_JOINTS_LOCATE_INFO_EXT };
     locateInfo.baseSpace = localSpace;
-    locateInfo.time = frameState.predictedDisplayTime;
+    locateInfo.time = predictedDisplayTime;
 
-    XrHandTrackingAimStateFB aimState { XR_TYPE_HAND_TRACKING_AIM_STATE_FB, XR_NULL_HANDLE, 0  };
     XrHandJointLocationsEXT jointLocations { XR_TYPE_HAND_JOINT_LOCATIONS_EXT };
     jointLocations.jointCount = XR_HAND_JOINT_COUNT_EXT;
     jointLocations.jointLocations = mHandJoints.data();
-    jointLocations.next = &aimState;
+    mGestureManager->populateNextStructureIfNeeded(jointLocations);
```

Comment:
Suggestion for future changes: Couldn't the Gesture Manager be the responsible of providing the whole XrHandJointLocationsEXT data structure ?

Reply:
Hmm yeah, I was not planning to do what it became the third commit, but yeah, with that one it does not make much sense to do all this work here. I'll modify it.